imap_processing/ialirt/l0/decom_ialirt.py

```diff
@@ -0,0 +1,99 @@
+import collections
+import logging
+
+import xarray as xr
+
+from imap_processing.decom import decom_packets
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+def generate_xarray(packet_file: str, xtce: str, time_keys=None):
+    """
+    Generate xarray from unpacked data.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+    xtce : str
+        Path to the XTCE packet definition file.
+    time_keys : dict
+        Keys used for creating xarray dataset
+
+    Returns
+    -------
+    xr.Dataset
+        A dataset containing the decoded data fields with 'time' as the coordinating
+        dimension.
+
+    Example Output:
+    ---------------
+    # This is an example of what the xarray dataset might look like
+    # after being processed by this function.
+
+    <xarray.Dataset>
+    Dimensions:       (SC_SCLK_SEC: 5)
+    Coordinates:
+      * SC_SCLK_SEC   (SC_SCLK_SEC) int64 322168 322169 322170 322171 322172
+    Data variables:
+        SC_MAG_STATUS (SC_SCLK_SEC) int64 0 1 0 1 0
+        SC_HIT_STATUS (SC_SCLK_SEC) int64 1 0 1 0 1
+
+    This example shows a dataset with 'SC_SCLK_SEC' as the coordinate
+    and two data variables 'SC_MAG_STATUS' and 'SC_HIT_STATUS'.
+    """
+    try:
+        packets = decom_packets(packet_file, xtce)
+    except Exception as e:
+        logger.error(f"Error during packet decomposition: {e!s}")
+        return
```

Comment:
```suggestion
    packets = decom_packets(packet_file, xtce)
```
I think we should actually fail here, so no need for a try/except.